[BUG]: RC3: Setup getting stuck at preprocessing pipeline

### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).


### Issue Description

Yesterday I upgraded to the latest release of pycaret (3.0.0rc3) from rc2. While it did fix a lot of bugs from previous version, I could not run `exp.setup(...)` with my dataset. This has not been an issue before. When running it gets stuck on some preprocessing step as you can see in the picture below.

![image](https://user-images.githubusercontent.com/107557528/178691805-2e71480f-1d07-4485-a124-7c61f3251c55.png)

As I write this, it has run for 10 min without continuing (setup usually takes maximum 1 min with my dataset). After interrupting the process, it turns out that I am stuck here:

![image](https://user-images.githubusercontent.com/107557528/178692724-843ebaa8-6e59-4ee0-a581-ee57a4aaf77c.png)

Unfortunately, I can not share my full dataset and can not reproduce the error on any of the open datasets. 

My dataset has shapes (4460,906) and has a lot of null values.

I understand that it is hard for you to find the error that occurs for me without my dataset. I will try to manipulate my dataset to upload it here. In the meantime, if you know what may be causing the issue, please tell me!

### Reproducible Example

```python
df = pd.read_csv('my_csv.csv')
exp = ClassificationExperiment()
exp.setup(data=df)
```


### Expected Behavior

Run as intentional.

### Actual Results

```python-traceback
Getting stuck in preprocessing as mentioned above.
```


### Installed Versions

<details>
3.0.0rc3
</details>

Sorry to bother you again, just mentioning that the same issue occurs when setting `remove_multicollinearity = True`

> 2022-07-13 13:05:53 - [ERROR] [main.<module>:119]: Traceback (most recent call last):
...
  File ".../lib/python3.8/site-packages/pycaret/classification/oop.py", line 850, in setup
    self.pipeline.fit(self.X_train, self.y_train)
  File ".../lib/python3.8/site-packages/pycaret/internal/pipeline.py", line 158, in fit
    _fit_one(self._final_estimator, X, y, **fit_params_last_step)
  File ".../lib/python3.8/site-packages/pycaret/internal/pipeline.py", line 36, in _fit_one
    transformer.fit(*args, **fit_params)
  File ".../lib/python3.8/site-packages/pycaret/internal/preprocess/transformers.py", line 209, in fit
    self.transformer.fit(*args, **fit_params)
  File ".../lib/python3.8/site-packages/pycaret/internal/preprocess/transformers.py", line 389, in fit
    data = X.merge(y.to_frame(), left_index=True, right_index=True)
  File ".../lib/python3.8/site-packages/pandas/core/frame.py", line 9351, in merge
    return merge(
  File ".../lib/python3.8/site-packages/pandas/core/reshape/merge.py", line 122, in merge
    return op.get_result()
  File ".../lib/python3.8/site-packages/pandas/core/reshape/merge.py", line 725, in get_result
    result_data = concatenate_managers(
  File ".../lib/python3.8/site-packages/pandas/core/internals/concat.py", line 221, in concatenate_managers
    values = values.copy()
numpy.core._exceptions._ArrayMemoryError: Unable to allocate 209. GiB for an array with shape (6292333, 4460) and data type float64


Now I have `low_variance_threshold=None`.

--- a/pycaret/internal/preprocess/transformers.py
+++ b/pycaret/internal/preprocess/transformers.py
@@ -139,7 +139,13 @@ class TransformerWrapper(BaseEstimator, TransformerMixin):
 
             # Add all derivative columns: cols that originate from another
             # and start with its progenitor name, e.g. one-hot encoded columns
-            columns.extend(list(df.columns[df.columns.str.startswith(f"{col}_")]))
+            columns.extend(
+                [
+                    c
+                    for c in df.columns
+                    if c.startswith(f"{col}_") and c not in original_df
+                ]
+            )
 
         # Add remaining new columns (non-derivatives)
         columns.extend([col for col in df if col not in columns])

